libs/model/src/middleware/auth.ts

```
@@ -414,6 +418,34 @@ export function authVerified() {
   };
 }
 
+/**
+ * Creates an authorization context for the actor when authenticated,
+ * but anonymous access is allowed.
+ * This is mainly used when querying communities with gating conditions.
+ */
+export async function authOptional(
+  ctx: Context<typeof AuthContextInput, typeof AuthContext>,
+) {
+  if (!ctx.actor.user || !ctx.actor.address) return;
+  if (ctx.payload.community_id === ALL_COMMUNITIES) return;
+
+  try {
+    const { address, is_author } = await findAddress(
+      ctx.actor,
+      ctx.payload.community_id,
+      ['admin', 'moderator', 'member'],
+    );
+
+    (ctx as { context: AuthContext }).context = {
+      address,
+      is_author,
+      community_id: ctx.payload.community_id,
+    };
+  } catch {
+    // ignore as optional
```

I think we should never completely ignore an error– it should at least be logged as a warning, or if it throws under the unauth case, then explicitly ignore that case while logging other cases.